macosx/Controller.m

```diff
@@ -3692,16 +3733,93 @@ - (NSRect) previewPanel: (QLPreviewPanel *) panel sourceFrameOnScreenForPreviewI
     }
 }
 
+- (IBAction) performTouchBarAction: (id) sender //senders here are always NSButtons, not NSTouchBarItems
+{
+    if (sender == createTouchBarButton)
+    {
+        [self createFile:sender];
+    }
+    if (sender == openFileTouchBarButton)
+    {
+        [self openShowSheet:sender];
+    }
+    if (sender == removeTouchBarButton)
+    {
+        [self removeNoDelete:sender];
+    }
+    if (sender == pauseAllTouchBarButton)
+    {
+        [self stopAllTorrents:sender];
+    }
+    if (sender == resumeAllTouchBarButton)
+    {
+        [self resumeAllTorrents:sender];
+    }
+    if (sender == speedLimitTouchBarButton)
+    {
+        [self toggleSpeedLimit: sender];
+    }
+    if (sender == cleanupTouchBarButton)
+    {
+        [self clearCompleted: sender];
+    }
+    if (sender == quickLookTouchBarButton)
+    {
+        [self toggleQuickLook:sender];
+    }
+    if (sender == showInfoTouchBarButton)
+    {
+        [self showInfo:sender];
+    }
+}
```

Comment:
sender can't be == to all of these, so this can be improved by being written as a switch-case.

In reality, you're doing a dictionary lookup from sender to method. I'm not an Obj-C dev, so I don't know the right idioms to use here, but ideally this should look something like:
```js
var implementations = {
  createTouchBarButton: createFile,
  openFileTouchBarButton: openShowSheet,
  /* ... */
  };
implementations[sender](self, sender);
```

The same applies to syncToTouchBarForItem.